src/iqm/iqm_client/models.py

```diff
@@ -529,6 +517,92 @@ class QuantumArchitecture(BaseModel):
     """Details about the quantum architecture."""
 
 
+class HeraldingMode(str, Enum):
+    """Heralding mode for circuit execution.
+
+    Heralding is the practice of generating data about the state of qubits prior to execution of a circuit.
+    This can be achieved by measuring the qubits immediately before executing each shot for a circuit."""
+
+    NONE = 'none'
+    """Do not do any heralding."""
+    ZEROS = 'zeros'
+    """Perform a heralding measurement after qubit initialization, only retain shots with an all-zeros result.
+
+    Note: in this mode, the number of shots returned after execution will be less or equal to the requested amount
+    due to the post-selection based on heralding data."""
+
+
+class MoveGateValidationMode(Enum):
```

Comment:
```suggestion
class MoveGateValidationMode(str, Enum):
```
Like HeraldingMode? `enum.StrEnum` would be better but it is only available in 3.11+

Reply:
Also the other enums.